tests/test_compat.py

```diff
@@ -323,3 +330,118 @@ def test_rotate_zeta():
+    np.testing.assert_allclose(surf_inner.R_lmn, eq_half_rho.surface.R_lmn, err_msg="R")
+    np.testing.assert_allclose(surf_inner.Z_lmn, eq_half_rho.surface.Z_lmn, err_msg="Z")
+    np.testing.assert_allclose(surf_inner.NFP, eq_half_rho.surface.NFP, err_msg="NFP")
+
+    # test geometry, profiles, |B| and |F| match btwn orig eq and new contracted eq
+    data_keys = ["|B|", "|F|", "R", "Z", "lambda", "p", "iota", "sqrt(g)"]
+    contract_grid = LinearGrid(
+        rho=np.linspace(0, 1.0, eq.L),
+        M=eq.M_grid,
+        N=eq.N_grid,
+        NFP=eq.NFP,
+    )
+    grid = LinearGrid(
+        rho=np.linspace(0, rho, eq.L),
+        M=eq.M_grid,
+        N=eq.N_grid,
+        NFP=eq.NFP,
+    )
+    contract_data = eq_half_rho.compute(data_keys, grid=contract_grid)
+    data = eq.compute(data_keys, grid=grid)
+    for key in data_keys:
+        if key != "|F|" and key != "sqrt(g)":
+            np.testing.assert_allclose(contract_data[key], data[key], err_msg=key)
+    np.testing.assert_allclose(
+        contract_data["|F|"], data["|F|"], rtol=2e-4, err_msg="|F|"
+    )
+    np.testing.assert_allclose(
+        contract_data["sqrt(g)"] * 2, data["sqrt(g)"], rtol=1e-8, err_msg="sqrt(g)"
+    )
+
+    ## test only the profiles and geometry for the eq without contracted profiles,
+    ## as the other quantities will be different since the profiles
+    ## now differ in real space
+    np.testing.assert_allclose(
+        surf_inner.R_lmn, eq_half_rho_same_profiles.surface.R_lmn, err_msg="R"
+    )
+    np.testing.assert_allclose(
+        surf_inner.Z_lmn, eq_half_rho_same_profiles.surface.Z_lmn, err_msg="Z"
+    )
+    np.testing.assert_allclose(
+        surf_inner.NFP, eq_half_rho_same_profiles.surface.NFP, err_msg="NFP"
+    )
+
+    # test geometry match btwn orig eq and new contracted eq
+    data_keys = ["R", "Z", "lambda"]
+    contract_grid = LinearGrid(
+        rho=np.linspace(0, 1.0, eq.L),
+        M=eq.M_grid,
+        N=eq.N_grid,
+        NFP=eq.NFP,
+    )
+    grid = LinearGrid(
+        rho=np.linspace(0, rho, eq.L),
+        M=eq.M_grid,
+        N=eq.N_grid,
+        NFP=eq.NFP,
+    )
+    contract_data = eq_half_rho_same_profiles.compute(data_keys, grid=contract_grid)
+    data = eq.compute(data_keys, grid=grid)
+    for key in data_keys:
+        np.testing.assert_allclose(contract_data[key], data[key], err_msg=key)
+    # test profiles
+    assert eq_half_rho_same_profiles.pressure.equiv(eq.pressure)
+    assert eq_half_rho_same_profiles.current.equiv(eq.current)
+
+
+@pytest.mark.unit
+def test_contract_equilibrium_warns_errors():
+    """Test contract_equilibrium utility function warning for profiles."""
+    eq = get("ESTELL")
+    rho = 0.5
+    eq.pressure += eq.pressure
+    with pytest.warns(UserWarning, match="SplineProfile"):
+        eq_half_rho = contract_equilibrium(eq, rho, profile_num_points=50)
+    eq.anisotropy = FourierZernikeProfile()
+    with pytest.raises(ValueError, match="FourierZernike"):
```

Comment:
This test is failing because it does not trigger the correct warning. You added an `anisotropy` profile, but the `pressure` profile gets checked first, so it gives the same warning as in the line above. The easiest solution would be to change this to `eq.pressure = FourierZernikeProfile()` instead of setting the anisotropy.